### ISSUE TITLE
Quirks of function reprs

### ISSUE THREAD

**Issue by oleobal** [oleobal/dd-dicetron#1]
Order doesn't seem deterministic: 

```
$ ./dd-dice 'max(d10,5)'
[9].max(5): 9
$ ./dd-dice 'max(d10,5)'
5.max([1]): 5
```
with `--explain`:
```
$ ./dd-dice 'max(d10,5)' --explain
d10->[6]
.max(5)->[6]
6
$ ./dd-dice 'max(d10,5)' --explain
5
.max(d10->[3])->5
5
```

**Comment by oleobal** [comment]
The reason for the improper order is that the `min` and `max` builtins would call `sort` on their arguments, which would sort them in-place. 

Other builtins probably have the same problem. `best` and `worst`, `sort` and `rsort`. They should create a copy.

**Comment by oleobal** [comment]
Fixed by b996f784b728a9480dca49267a3ab2fbc76b34d6

Except for sort/rsort which are unimportant, I just removed them.
